### PR TITLE
Revoke Credentials and Refactor

### PR DESCRIPTION
**Pull request by sslaws**
# Fixes or Implements [AB#10624](https://dev.azure.com/qslvic/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10624)

-   [X] Enhancement
-   [ ] Bug
-   [ ] Documentation

## Description

Refactors the code - Connection is operational
Adds initial Revoke Server side code
Adds Create Credentials as per new design

I also merged from Dev hence the dependbot updates.

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

-   [X] Unit Tests Updated
-   [ ] Functional Tests Updated
-   [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

Yes

### Browsers Tested

-   [ ] Chrome
-   [ ] Safari
-   [ ] Edge
-   [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant. Include any specialized deployment steps here.

### CODE REVIEW

File: Apps/Common/src/Delegates/RestWalletIssuerDelegate.cs [hunk 1]
```diff
@@ -202,15 +202,97 @@ public async Task<RequestResult<CredentialResponse>> CreateCredentialAsync<T>(Wa
...
+                    retVal.ResultError = new RequestResultError() { ResultMessage = $"Unable to connect to AcaPy Agent, HTTP Error {response.StatusCode}", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.WalletIssuer) };
+                    this.logger.LogError($"Unable to connect to endpoint {endpoint}, HTTP Error {response.StatusCode}\n{payload}");
+                }
+            }
+#pragma warning disable CA1031 // Do not catch general exception types
+            catch (Exception e)
+            {
+                retVal.ResultError = new RequestResultError() { ResultMessage = $"Exception Deleting Connection: {e}", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.WalletIssuer) };
+                this.logger.LogError($"Unexpected exception in DisconnectConnectionAsync {e}");
+            }
```
Comment (MikeLyttle) [inline Apps/Common/src/Delegates/RestWalletIssuerDelegate.cs:236]: Nit: missing `#pragma warning restore CA1031`.

File: Apps/Common/src/Delegates/RestWalletIssuerDelegate.cs [hunk 2]
```diff
@@ -202,15 +202,97 @@ public async Task<RequestResult<CredentialResponse>> CreateCredentialAsync<T>(Wa
...
+                        retVal.ResultError = new RequestResultError() { ResultMessage = $"Unable to connect to AcaPy Agent, HTTP Error {response.StatusCode}", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.WalletIssuer) };
+                        this.logger.LogError($"Unable to connect to endpoint {endpoint}, HTTP Error {response.StatusCode}\n{payload}");
+                    }
+                }
+    #pragma warning disable CA1031 // Do not catch general exception types
+                catch (Exception e)
+                {
+                    retVal.ResultError = new RequestResultError() { ResultMessage = $"Exception Revoking credential: {e}", ErrorCode = ErrorTranslator.ServiceError(ErrorType.CommunicationExternal, ServiceType.WalletIssuer) };
+                    this.logger.LogError($"Unexpected exception in RevokeCredentialAsync {e}");
+                }
```
Comment (MikeLyttle) [inline Apps/Common/src/Delegates/RestWalletIssuerDelegate.cs:284]: Nit: missing `#pragma warning restore CA1031`.

File: Apps/WebClient/src/Server/Controllers/WalletController.cs [hunk 3]
```diff
@@ -88,21 +87,67 @@ public ActionResult GetConnection(string hdId)
...
+        [HttpDelete]
+        [Route("{hdid}/Connection/{connectionId}")]
+        [Authorize(Policy = UserProfilePolicy.Write)]
+        public ActionResult DisconnectConnection(string hdId, Guid connectionId)
+        {
+            throw new NotSupportedException();
+        }
+
+        /// <summary>
+        /// Updates the Health Gateway Wallet Credentail and revokes the credential with the Agent.
```
Comment (MikeLyttle) [inline Apps/WebClient/src/Server/Controllers/WalletController.cs:104]: Typo: `Credentail` should be `Credential` (throughout file).

File: Apps/WebClient/src/Server/Controllers/WalletController.cs [hunk 4]
```diff
@@ -88,21 +87,67 @@ public ActionResult GetConnection(string hdId)
...
+        /// <param name="targetId">The target id of the immunization to create a credential for.</param>
+        /// <returns>Something.</returns>
+        [HttpPost]
         [Route("{hdid}/Credential")]
-        [Authorize(Policy = UserProfilePolicy.Read)]
-        public ActionResult GetCredential(string hdId, Guid exchangeId)
+        [Authorize(Policy = UserProfilePolicy.Write)]
+        public async Task<ActionResult> CreateCredential(string hdId, [FromBody] string targetId)
+        {
+            this.logger.LogDebug($"Creating credenial for user {hdId}");
```
Comment (MikeLyttle) [inline Apps/WebClient/src/Server/Controllers/WalletController.cs:114]: Typo: `credenial` should be `credential` (throughout file).

File: Apps/WebClient/src/appsettings.json [hunk 5]
```diff
@@ -91,25 +91,28 @@
...
-        "ZipkinUri": "",
-        "IgnorePathPrefixes": ["/health"]
-    },
-    "AcaPy": {
-        "agentApiUrl": "***",
-        "agentApiKey": "***",
-        "webhookApiKey": "***",
-        "schemaName": "Immunization",
-        "schemaVersion": "1.0"
+        "ContentSecurityPolicy": {
```
Comment (MikeLyttle) [inline Apps/WebClient/src/appsettings.json:97]: Lint: indentation is off.

**Review by MikeLyttle — APPROVED** [review]

**Review by FuriousLlama — APPROVED** [review]

**Pull request merged**